src/coreclr/src/gc/gcdesc.h

```diff
@@ -199,28 +199,38 @@ class CGCDesc
     static size_t GetNumPointers (MethodTable* pMT, size_t ObjectSize, size_t NumComponents)
     {
         size_t NumOfPointers = 0;
-        CGCDesc* map = GetCGCDescFromMT(pMT);
-        CGCDescSeries* cur = map->GetHighestSeries();
-        ptrdiff_t cnt = (ptrdiff_t) map->GetNumSeries();
 
-        if (cnt > 0)
+        if (pMT->ContainsPointers())
         {
-            CGCDescSeries* last = map->GetLowestSeries();
-            while (cur >= last)
+            CGCDesc* map = GetCGCDescFromMT(pMT);
+            CGCDescSeries* cur = map->GetHighestSeries();
+            ptrdiff_t cnt = (ptrdiff_t)map->GetNumSeries();
+
+            if (cnt >= 0)
```

Comment:
My analysis indicates the change here to change from > to >= is a small performance optimization. From my analysis cnt == 0 can't actually happen so this is ok, but I want to verify that was the rationale.

Reply:
My primary motivation for this was to make the conditions and flow to be more similar to the `go_through_object` macros, so that easier to validate that the logic in this method matches it.

You are right that the side-effect is minor perf optimization.